Source/ARTClientOptions.m

```diff
@@ -66,7 +65,7 @@ - (NSString*)getRestHost {
     if ([_environment isEqualToString:ARTDefaultProduction]) {
         return [ARTDefault restHost];
     }
-    return _environment ? [NSString stringWithFormat:@"%@-%@", _environment, [ARTDefault restHost]] : [ARTDefault restHost];
+    return self.hasEnvironment ? [NSString stringWithFormat:@"%@-%@", _environment, [ARTDefault restHost]] : [ARTDefault restHost];
```

Comment:
Consider encapsulating this `[NSString stringWithFormat:@"%@-%@", _environment, [ARTDefault restHost]]` as a Helper/Util/Category

Reply:
@lukasz-szyszkowski can be [ARTDefault envRestHost]

Reply:
Also why getter is `getRestHost` instead of `restHost`?

Reply:
it's declared here:
```objective-c
@property (readwrite, strong, nonatomic, getter=getRestHost) NSString *restHost;
@property (readwrite, strong, nonatomic, getter=getRealtimeHost) NSString *realtimeHost;
```
but since we can't set it from outside, IMO it should be:
```objective-c
@property (nonatomic, readonly) NSString *restHost;
@property (nonatomic, readonly) NSString *realtimeHost;
```

Reply:
changed here 501c48d3c099432d213f13cbc51337fe308af6d2